### ISSUE TITLE
publish.single() fails on TLS

### ISSUE THREAD

**Issue by pcanterino** [eclipse/paho.mqtt.python#180]
I tried to publish a single message using publish.single(). It works as expected if there is no TLS encryption:

```
import paho.mqtt.publish as p
p.single("test", payload="Hello!")
```

If i try to send the message using TLS encryption, it doesn't work:

`p.single("test", payload="Hello!", port=8883, tls={'ca_certs': '/path/to/ca'})`

I get the following stacktrace:

```
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/paho_mqtt-1.2.2-py3.5.egg/paho/mqtt/publish.py", line 224, in single
    multiple([msg], hostname, port, client_id, keepalive, will, auth, tls, protocol, transport)
  File "/usr/local/lib/python3.5/dist-packages/paho_mqtt-1.2.2-py3.5.egg/paho/mqtt/publish.py", line 175, in multiple
    client.connect(hostname, port, keepalive)
  File "/usr/local/lib/python3.5/dist-packages/paho_mqtt-1.2.2-py3.5.egg/paho/mqtt/client.py", line 700, in connect
    return self.reconnect()
  File "/usr/local/lib/python3.5/dist-packages/paho_mqtt-1.2.2-py3.5.egg/paho/mqtt/client.py", line 835, in reconnect
    ciphers=self._tls_ciphers)
  File "/usr/lib/python3.5/ssl.py", line 1069, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib/python3.5/ssl.py", line 686, in __init__
    self._context = SSLContext(ssl_version)
  File "/usr/lib/python3.5/ssl.py", line 361, in __new__
    self = _SSLContext.__new__(cls, protocol)
TypeError: an integer is required (got type NoneType)
```

It's not a problem of this machine, because it works if I setup the MQTT connection manually using the object oriented interface.

I can't tell if it's a bug or if it's me doing something wrong.

EDIT: Fixed formatting

**Comment by pcanterino** [comment]
I recently tested subscribe.simple() using TLS: Works without a problem.

```
import paho.mqtt.subscribe as s
msg = s.simple("test", port=8883, tls = {'ca_certs': '/path/to/ca'})
print(msg.payload)
```

-> `b'Hello!'`

**Comment by PierreF** [comment]
It worked using a released version of paho-mqtt ? or using the develop branch ?

I can still reproduce the bug in fixes branch (or latest paho 1.2.2). It's due to the way publish.multiple fallback and no tls_version is provided: [it fallback to None](https://github.com/eclipse/paho.mqtt.python/blob/a419f438cb4c60b57351b687d55eff1299cb434b/src/paho/mqtt/publish.py#L167).
And this fallback is keep as-is with the tls_set function.

On develop, this is fixed by PR #92 that changed the way tls_set was called.

**Comment by pcanterino** [comment]
I tested it on release 1.2.2. publish.single() does not work but subscribe.simple() works. I wanted to point out that it is not a problem with the TLS library on my machine.

So if this is fixed on develop, i will close this issue.

**Comment by PierreF** [comment]
Ok, didn't see the subscribe vs publish difference :)

I've committed also a fix on fixes branch (commit d38c5f8).
